### PR TITLE
fixed: due to RAII usage in libcereal archive must go out of scope

### PR DESCRIPTION
**Pull request by akva2**
if not, the stream is not finished when we read out the data

### CODE REVIEW

File: src/SIM/SIMsolution.C [hunk 1]
```diff
@@ -46,9 +46,11 @@ bool SIMsolution::saveSolution (SerializeMap& data,
...
 #ifdef HAS_CEREAL
   std::ostringstream str;
-  cereal::BinaryOutputArchive archive(str);
-  for (const Vector& sol : this->getSolutions())
-    archive(sol);
+  {
+    cereal::BinaryOutputArchive archive(str);
+    for (const Vector& sol : this->getSolutions())
+      archive(sol);
+  }
```
Comment (kmokstad) [inline src/SIM/SIMsolution.C]: It does not have an explicit `close`method one can use instead?

Comment (akva2) [reply src/SIM/SIMsolution.C]: unfortunately no.

Comment (akva2) [reply src/SIM/SIMsolution.C]: the alternative is new / delete.

**Comment by akva2** [comment]
jenkins build this with downstreams ifem-advectiondiffusion=24 ifem-stokes=105 ifem-boussinesq=4 please

**Comment by akva2** [comment]
jenkins build this with downstreams ifem-advectiondiffusion=24 ifem-stokes=105 ifem-boussinesq=4 please

**Comment by akva2** [comment]
okay to merge ?

**Pull request merged**